proposals/913-components-repo/README.md

```diff
@@ -0,0 +1,497 @@
+2. Backport historical component versions; only actively maintained assets will be migrated.
+3. Remove the existing `components/` directory in the main KFP repo without notice. We will announce the new repository,
+   point contributors at the new onboarding documentation, and give the community a minimum of 90 days to move their
+   assets before deleting the legacy directory.
+4. Vendor specific components. Vendors are highly encouraged to use the same repository structure in their own
+   repository.
+
+## Proposal
+
+### Repository Layout
+
+Create a new repository (e.g. `kubeflow/kfp-components`) with the following top-level layout:
+
+```text
+root
+├── components
+│   ├── README.md (catalog landing page)
+│   ├── __init__.py (auto-imports all components for clean `from kfp_components import training` usage)
+│   ├── training/
+│   │   ├── README.md (category index listing each component with summaries/links)
+│   │   ├── __init__.py (re-exports all components in this category)
+│   │   └── <component-name>/
+│   │       ├── __init__.py (exposes the component entrypoint for imports)
+│   │       ├── component.py
+│   │       ├── metadata.yaml
+│   │       ├── README.md
+│   │       ├── OWNERS
+│   │       ├── example_pipelines.py
+│   │       ├── tests/
+│   │       │   └── test_component.py
+│   │       └── <supporting_files>
+│   └── ... (other categories: evaluation/, data_processing/, etc.)
+├── pipelines
+│   ├── README.md (catalog landing page)
+│   ├── __init__.py (auto-imports all pipelines for `from kfp_components import pipelines.training` usage)
+│   ├── training/
+│   │   ├── README.md (category index listing each pipeline with summaries/links)
+│   │   ├── __init__.py (re-exports all pipelines in this category)
+│   │   └── <pipeline-name>/
+│   │       ├── __init__.py (exposes the pipeline entrypoint)
+│   │       ├── pipeline.py
+│   │       ├── metadata.yaml
+│   │       ├── README.md
+│   │       ├── OWNERS
+│   │       ├── tests/
+│   │       │   └── test_pipeline.py
+│   │       └── <supporting_files>
+│   └── ... (other categories: evaluation/, data_processing/, etc.)
+├── third_party
+│   ├── pyproject.toml (third-party package build definition)
+│   ├── __init__.py
+│   ├── components
+│   │   ├── README.md (catalog landing page)
+│   │   ├── __init__.py
+│   │   └── ... (mirrors `components/` structure)
+│   └── pipelines
+│       ├── README.md (catalog landing page)
+│       ├── __init__.py
+│       └── ... (mirrors `pipelines/` structure)
+├── docs/
+│   ├── ONBOARDING.md
+│   ├── CONTRIBUTING.md
+│   ├── AGENTS.md
+│   ├── GOVERNANCE.md
+│   └── BESTPRACTICES.md
+├── scripts/
+│   ├── validate_metadata.py
+│   ├── generate_readme.py
+│   ├── update_init_imports.py (auto-generates category `__init__.py` import stubs; enforced by CI)
+│   └── verify_dependencies.py
+└── pyproject.toml (core package)
+```
+
+#### Core (`components/` and `pipelines/`)
```

Comment:
What are the differences between `components` and `pipelines` ?

Reply:
Components are the equivalent of a single Python function and pipelines are multiple components stitched together. KFP supports nested pipelines so you can treat a pipeline as a component.